src/libraries/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -803,14 +803,7 @@ public HttpListenerContext EndGetContext(IAsyncResult asyncResult)
                 if (authorizationHeader != null && (authenticationScheme & ~AuthenticationSchemes.Anonymous) != AuthenticationSchemes.None)
                 {
                     // Find the end of the scheme name.  Trust that HTTP.SYS parsed out just our header ok.
-                    for (index = 0; index < authorizationHeader.Length; index++)
-                    {
-                        if (authorizationHeader[index] == ' ' || authorizationHeader[index] == '\t' ||
-                            authorizationHeader[index] == '\r' || authorizationHeader[index] == '\n')
-                        {
-                            break;
-                        }
-                    }
+                    index = authorizationHeader.AsSpan().IndexOfAny(" \t\r\n");
```

Comment:
Index becomes negative if there is no match. We probably need to adjust condition bellow.